compiler/src/dotty/tools/dotc/semanticdb/TypeOps.scala

```diff
@@ -233,12 +233,8 @@ class TypeOps:
           val ssym = sym.symbolName
           s.TypeRef(spre, ssym, Seq.empty)
 
-        case TypeRef(pre, sym: Name) =>
-          // val spre = if tpe.hasTrivialPrefix then s.Type.Empty else loop(pre)
-          // Craft semanticdb symbol
-          // prefix symbol (e.g. "scala/") + name (e.g. "Nothing") + Global symbol suffix ("#")
-          // see: https://scalameta.org/docs/semanticdb/specification.html#symbol
-          val ssym = s"${pre.typeSymbol.symbolName}${sym.mangledString}#"
+        case tr @ TypeRef(_, _: Name) =>
+          val ssym = tr.symbol.symbolName
           s.TypeRef(s.Type.Empty, ssym, Seq.empty)
```

Comment:
Can we also try to compute the type of the prefix in case is is not empty?

Reply:
also this could be merged with the branch above - `tr.symbol` will just return the symbol if there already is one

Reply:
> also perhaps this could be merged with the branch above

Right!

https://github.com/lampepfl/dotty/pull/14608/commits/26939833920d3c64715fca030ff38cb1436d21dc